lib/graphql/client/schema.rb

```diff
@@ -66,14 +66,14 @@ def normalize_type_name(type_name)
         end
       end
 
-      def self.generate(schema)
+      def self.generate(schema, raise_on_unknown_enum_values:)
         mod = Module.new
         mod.extend ClassMethods
 
         cache = {}
         schema.types.each do |name, type|
           next if name.start_with?("__")
-          if klass = class_for(schema, type, cache)
+          if klass = class_for(schema, type, cache, raise_on_unknown_enum_values: raise_on_unknown_enum_values)
```

Comment:
Any thoughts on defining `raise_on_unknown_enum_value` on `mod`? I believe it could be accessed in `EnumType` via `schema_module`, saving the need to pass it around `class_for`.